app/models/locations/location.rb

```diff
@@ -153,10 +156,27 @@ def available_coordinates(_start, _number)
     []
   end
 
-  def remove_all_labwares
+  def remove_all_labwares(current_user)
     return if has_child_locations?
 
+    # audit that uer removed the location
+    self.create_audit(current_user, Audit::REMOVED_ALL_ACTION)
+
+    # copy array of labwares to be deleted (labwares will be empty after delete_all)
+    labwares_copy = []
```

Comment:
is there a better way to copy this array of labwares?
after labwares.delete_all it is emptied so cannot use the original

Reply:
dup and clone didn't seem to behave